### PR TITLE
Fix: Added check for null workcontacts map

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
#### List of Changes
- Added check for null workcontacts map

#### Motivation and Context
We receive a nullpointer when pdv returns null workcontact map

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserServiceImpl.java [hunk 1]
```diff
@@ -345,11 +345,23 @@ public Uni<String> createOrUpdateUserByFiscalCode(CreateUserDto userDto, LoggedU
...
         log.info("Updating user on userRegistry and userInstitution");
-        String mailUuid = userUtils.getMailUuidFromMail(userResource.getWorkContacts(), userDto.getUser().getInstitutionEmail())
-                .orElse(randomMailId.get());
+        String mailUuid;
+        if(!CollectionUtils.isNullOrEmpty(userResource.getWorkContacts())) {
+          mailUuid = userUtils.getMailUuidFromMail(userResource.getWorkContacts(), userDto.getUser().getInstitutionEmail())
+                    .orElse(randomMailId.get());
+        }else {
+            mailUuid = randomMailId.get();
+        }
```
Comment (manuraf) [inline apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserServiceImpl.java:354]: ditto
```
String mailUuid = Optional.ofNullable(userResource.getWorkContacts())
                .flatMap(contacts -> userUtils.getMailUuidFromMail(contacts, userDto.getUser().getInstitutionEmail()))
                .orElseGet(randomMailId);
```

**Review by manuraf — COMMENTED** [review]
it might be better to add some unit test about workContracts is null

**Review by manuraf — DISMISSED** [review]
